From 3d1767186da60c29086f61be24f8292876afa871 Mon Sep 17 00:00:00 2001
From: Eugene Kliuchnikov <eustas.ru@gmail.com>
Date: Tue, 30 Jul 2019 10:01:21 +0200
Subject: [PATCH] Fix include for EMCC build (#765)

---
 c/common/platform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/c/common/platform.h b/c/common/platform.h
index b6d634e..2633e04 100755
--- a/c/common/platform.h
+++ b/c/common/platform.h
@@ -29,7 +29,7 @@
 #include <brotli/port.h>
 #include <brotli/types.h>
 
-#if defined(OS_LINUX) || defined(OS_CYGWIN)
+#if defined(OS_LINUX) || defined(OS_CYGWIN) || defined(__EMSCRIPTEN__)
 #include <endian.h>
 #elif defined(OS_FREEBSD)
 #include <machine/endian.h>
-- 
2.6.1.windows.1

